pyproject.toml

```diff
@@ -1,3 +1,35 @@
+[project]
+name = "s2geometry"
+description = "Computational geometry and spatial indexing on the sphere"
+authors = [
+    { name = "Dan Larkin-York", email = "dan@arangodb.com" },
+    { name = "Google Inc."},
+    { name = "Koordinates Limited"},
+    { name = "Mike Playle", email = "mike@mythik.co.uk" },
+    { name = "Tiago Brito", email = "tiago.brito@90poe.io" },
+]
```

Comment:
how's this?

Reply:
@jmr Overall, this looks good to me. Thanks @zacharyburnett this is a nice update to more current practice. @jmr Does the Python wheel get built and tested in CI? If not, I would recommend adding this (though not part of this PR). If so, it would be nice to see whether that works with this new method (I would imagine the CI script needs to be updated).

Reply:
I'd recommend using `cibuildwheel` for that but you're right, out of scope of this PR

Reply:
There's currently no CI. :(